utils/network.go

```diff
@@ -0,0 +1,119 @@
+package utils
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/containernetworking/cni/pkg/ip"
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/vishvananda/netlink"
+)
+
+var slash32 = net.CIDRMask(32, 32)
+
+// DoNetworking performs the networking for the given config and IPAM result
+func DoNetworking(args *skel.CmdArgs, conf NetConf, result *types.Result) (string, string, error) {
+	hostVethName, contVethMac, err := setupContainerVeth(args.Netns, args.IfName, conf.MTU, result)
+	if err != nil {
+		return "", "", err
+	}
+
+	// Select the first 11 characters of the containerID for the host veth
+	newHostVethName := "cali" + args.ContainerID[:min(11, len(args.ContainerID))]
+	if err = setupHostVeth(hostVethName, newHostVethName); err != nil {
+		return "", "", err
+	}
+	return newHostVethName, contVethMac, nil
+}
+
+func setupContainerVeth(netns, ifName string, mtu int, res *types.Result) (string, string, error) {
+	var hostVethName, contVethMAC string
+	err := ns.WithNetNSPath(netns, func(hostNS ns.NetNS) error {
+		hostVeth, contVeth, err := ip.SetupVeth(ifName, mtu, hostNS)
+
+		if err != nil {
+			return err
+		}
+
+		// Handle IPv4
+		if res.IP4 != nil {
+			// Add a connected route to a dummy next hop so that a default route can be set
```

Comment:
I think this code used to live in libcalico, yeah?  Can we add a TODO to your list to migrate this to libcalico so that all integrations can configure container routes in the same way?